[TRTLLM-8714][fix] update create_input_processor to handle custom checkpoint format

Funatiq opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added support for specifying a checkpoint format during input processing, enabling smoother use of non-Hugging Face and multimodal checkpoints.
  * Introduced an optional configuration to bypass auto-loading of model configs when not needed, improving robustness.

* **Documentation**
  * Updated usage to highlight the new optional checkpoint format setting for input processing.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!--
Please write the PR title by following this template:

**[JIRA ticket/NVBugs ID/GitHub issue/None][type] Summary**

Valid ticket formats:
  - JIRA ticket: [TRTLLM-1234] or [FOOBAR-123] for other FOOBAR project
  - NVBugs ID: [https://nvbugs/1234567]
  - GitHub issue: [#1234]
  - No ticket: [None]

Valid types (lowercase): [fix], [feat], [doc], [infra], [chore], etc.

Examples:
  - [TRTLLM-1234][feat] Add new feature
  - [https://nvbugs/1234567][fix] Fix some bugs
  - [#1234][doc] Update documentation
  - [None][chore] Minor clean-up

Alternative (faster) way using CodeRabbit AI:

**[JIRA ticket/NVBugs ID/GitHub issue/None] @coderabbitai title**

NOTE: "@coderabbitai title" will be replaced by the title generated by CodeRabbit AI, that includes the "[type]" and title.
For more info, see /.coderabbit.yaml.

-->

## Description

* Modified the create_input_processor function to accept a checkpoint_format parameter, defaulting to "HF".
* The function now conditionally attempts to load the model configuration based on the specified format.

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- The reviewers assigned automatically/manually are appropriate for the PR.


- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

Funatiq committed:
[None][fix] update create_input_processor to handle custom checkpoint format

- Modified the create_input_processor function to accept a checkpoint_format parameter, defaulting to "HF".
- Add detailed parameter descriptions and return type clarification.
- The function now conditionally attempts to load the model configuration based on the specified format.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #19014](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/19014/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #14259](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/14259/) completed with status: 'FAILURE'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #19166](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/19166/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #19166](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/19166/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #14385](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/14385/) completed with status: 'SUCCESS'
Pipeline passed with automatic retried tests. Check the [rerun report](https://urm.nvidia.com/artifactory/sw-tensorrt-generic/llm-artifacts//LLM/main/L0_MergeRequest_PR/14385/test-results/rerun_report.html) for details.

Funatiq commented:
@corerabbitai review

Funatiq committed:
chore: Enhance error handling in create_input_processor

- Added debug logging for exceptions when loading the HF model configuration.
- Included a fallback message when skipping the HF config load based on checkpoint format.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
[None][fix] Enhance MultimodalEncoder to support custom checkpoint loading

- Get checkpoint_format in MultimodalEncoder and pass it to create_input_processor.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #21481](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/21481/) [ run ] triggered by Bot

Funatiq committed:
[None][docs] Update checkpoint_format and checkpoint_loader documentation in LLMArgs

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented on tensorrt_llm/inputs/registry.py:
```diff
@@ -408,20 +408,40 @@ def wrapper(model_cls: N) -> N:
-    except (ValueError, EnvironmentError):
-        config = None
+
+    if checkpoint_format is None or checkpoint_format == "HF":
```
Updated the description of `checkpoint_format` and `checkpoint_loader`.

Funatiq replied:
Had to revert the changes because of test issues.

Funatiq commented:
/bot run --disable-fail-fast

Funatiq commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #21897](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/21897/) [ run ] triggered by Bot. Commit: `dad1982`

Superjomn commented:
/bot run

tensorrt-cicd commented:
[PR_Github #21907](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/21907/) [ run ] triggered by Bot. Commit: `dad1982`

tensorrt-cicd commented:
[PR_Github #22050](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/22050/) [ run ] triggered by Bot. Commit: `f288868`

tensorrt-cicd commented:
[PR_Github #22032](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/22032/) [ run ] completed with state `ABORTED`. Commit: `dad1982` 
<a href="https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/16611/" target="_blank">LLM/main/L0_MergeRequest_PR #16611</a> <a href="https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/16611//display/redirect" target="_blank">(Blue Ocean)</a> completed with status: ABORTED

Funatiq commented:
/bot run

hchings commented on tensorrt_llm/llmapi/llm_args.py:
```diff
@@ -2528,7 +2534,12 @@ class TorchLlmArgs(BaseLlmArgs):
+        description=
+        "The format of the provided checkpoint. You may use a custom checkpoint format by subclassing "
+        "`BaseCheckpointLoader` and registering it with `register_checkpoint_loader`.\n"
+        "If neither checkpoint_format nor checkpoint_loader are provided, checkpoint_format will be set to HF "
```
nit: can replace `HF` with `HF (Hugging Face)` for clarity. Same for L2525.

hchings reviewed (approved):
Left a nit. Otherwise LGTM.
